### PR TITLE
refactor PointRend mask head

### PR DESCRIPTION
**Pull request by bowenc0221**

### CODE REVIEW

**Comment by ppwwyyxx** [comment]
The merge seems broken: `mask_head.py` now contains many unused methods.

Also, `roi_mask_point_loss` needs to be changed given that label sampling logic is moved out.

File: projects/PointRend/point_rend/mask_head.py [hunk 1]
```diff
@@ -46,32 +50,32 @@ def calculate_uncertainty(logits, classes):
...
 
         # fmt: off
-        self.num_classes            = cfg.MODEL.ROI_HEADS.NUM_CLASSES
-        conv_dim                    = cfg.MODEL.ROI_MASK_HEAD.CONV_DIM
-        self.fc_dim                 = cfg.MODEL.ROI_MASK_HEAD.FC_DIM
-        num_fc                      = cfg.MODEL.ROI_MASK_HEAD.NUM_FC
-        self.output_side_resolution = cfg.MODEL.ROI_MASK_HEAD.OUTPUT_SIDE_RESOLUTION
         self.input_channels         = input_shape.channels
         self.input_h                = input_shape.height
         self.input_w                = input_shape.width
```
Comment (ppwwyyxx) [inline projects/PointRend/point_rend/mask_head.py:77]: please remove `self.` - it seems this is not needed

**Comment by bowenc0221** [comment]
Fixed, the PointRend now contains the new logic as we discussed

```python
class PointRend:
    def forward(features, instances):
        if self.training:
            proposal_boxes = [x.proposal_boxes for x in instances]
            coarse_mask = self.coarse_head(self._roi_pooler(features, proposal_boxes))
            losses = mask_rcnn_loss(coarse_mask, instances)

            point_coords, point_labels = self._sample_train_points(coarse_mask, instances)
            point_fine_grained_features = self._point_pooler(features, proposal_boxes, point_coords)
            point_logits = self._get_point_logits(point_fine_grained_features, point_coords, coarse_mask)
            losses["loss_mask_point"] = roi_mask_point_loss(point_logits, instances, point_labels)
            return losses
        else:
            pred_boxes = [x.pred_boxes for x in instances]
            coarse_mask = self.coarse_head(self._roi_pooler(features, pred_boxes)
            return self._subdivision_inference(features, coarse_mask, instances)

    def _roi_pooler: shared, see https://github.com/facebookresearch/detectron2/pull/2772
    def _point_pooler: shared, similar to roi_pooler
    def _subdivision_inference: shared, same as the inference part of current _forward_mask_point.
    def _sample_train_points: unique
    def _get_point_logits: different
```
